java/code/src/com/redhat/rhn/manager/content/ubuntu/UbuntuErrataManager.java

```diff
@@ -116,22 +117,40 @@ private static Optional<String> archToPackageArchLabel(String arch) {
 
     private static Map<String, UbuntuErrataInfo> downloadUbuntuErrataInfo(String jsonDBUrl) throws IOException {
         HttpClientAdapter httpClient = new HttpClientAdapter();
-        HttpGet httpGet = new HttpGet(jsonDBUrl);
+        String bzipJsonDBUrl = jsonDBUrl + ".bz2";
+        HttpGet httpGet = new HttpGet(bzipJsonDBUrl);
         LOG.info("download ubuntu errata start");
         HttpResponse httpResponse = httpClient.executeRequest(httpGet);
         int statusCode = httpResponse.getStatusLine().getStatusCode();
         if (statusCode == HttpStatus.SC_OK) {
             try (
                 InputStream responseStream = httpResponse.getEntity().getContent();
-                Reader responseReader = new InputStreamReader(responseStream);
+                BZip2CompressorInputStream bzIn = new BZip2CompressorInputStream(responseStream);
+                Reader responseReader = new InputStreamReader(bzIn);
             ) {
                 Map<String, UbuntuErrataInfo> errataInfo = GSON.fromJson(responseReader, ERRATA_INFO_TYPE);
                 LOG.info("download ubuntu errata end");
                 return errataInfo;
             }
         }
         else {
-            throw new IOException("error downloading " + jsonDBUrl + " status code " + statusCode);
+            LOG.info("Failed to get bzip2 DB - try plain DB");
+            httpGet = new HttpGet(jsonDBUrl);
```

Comment:
Small nitpick . Some of the code is duplicate in if/else claus. Maybe we can extract that in a separate method.

Reply:
Don't know a good way to do this without making this code unreadable. Feel free to provide a solution.